### PR TITLE
List patterns: Fix pattern lookup on nullable types

### PR DESCRIPTION
**Pull request by alrz**
Test plan https://github.com/dotnet/roslyn/issues/51289

### CODE REVIEW

**Review by 333fred — APPROVED** [review]

File: src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests_ListPatterns.cs [hunk 1]
```diff
@@ -1520,6 +1520,62 @@ class D
...
+
+    public static bool Test(S? s)
+    {
+        return s is [42];
+    }
+
+    public static void Main()
+    {
+        Console.WriteLine(Test(new()));
+        Console.WriteLine(Test(default));
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests_ListPatterns.cs:1541]: nit: let's use `null` and `new S()` to be completely unambiguous

**Review by jcouv — COMMENTED** [review]
Done with review pass (iteration 2). Only a small test suggestion

File: src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests_ListPatterns.cs (hunk 1 above)
Comment (alrz) [reply src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests_ListPatterns.cs:1541]: We're not testing `new()` here though 😄 I will take your suggestion anyways.

**Comment by alrz** [comment]
(CI is not running for some reason)

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 4)

**Pull request merged**